src/ray/raylet/node_manager.cc

```diff
@@ -551,14 +551,12 @@ ray::Status NodeManager::RegisterGcs() {
         RAY_CHECK_OK(gcs_client_->Nodes().AsyncCheckSelfAlive(
             // capture checking ptr here because vs17 fail to compile
             [checking_ptr = &checking](auto status, auto alive) mutable {
-              if (status.ok()) {
-                if (!alive) {
-                  // GCS think this raylet is dead. Fail the node
-                  RAY_LOG(FATAL)
-                      << "GCS consider this node to be dead. This may happen when "
-                      << "GCS is not backed by a DB and restarted or there is data loss "
-                      << "in the DB.";
-                }
+              if ((status.ok() && !alive) || status.IsAuthError()) {
```

Comment:
I think AuthError needs a different fatal message, it's different from the existing case where the node belongs to this cluster but it's dead.

I guess when AuthError happens, there must be a bug right

Reply:
done